src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFDisksCollector.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.collectors.telemetry;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.performanceanalyzer.OpenSearchResources;
+import org.opensearch.performanceanalyzer.commons.OSMetricsGeneratorFactory;
+import org.opensearch.performanceanalyzer.commons.collectors.DisksCollector;
+import org.opensearch.performanceanalyzer.commons.collectors.PerformanceAnalyzerMetricsCollector;
+import org.opensearch.performanceanalyzer.commons.metrics.AllMetrics;
+import org.opensearch.performanceanalyzer.commons.metrics.MetricsConfiguration;
+import org.opensearch.performanceanalyzer.commons.metrics_generator.DiskMetricsGenerator;
+import org.opensearch.performanceanalyzer.commons.metrics_generator.OSMetricsGenerator;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatExceptionCode;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatMetrics;
+import org.opensearch.telemetry.metrics.Histogram;
+import org.opensearch.telemetry.metrics.MetricsRegistry;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+public class RTFDisksCollector extends PerformanceAnalyzerMetricsCollector {
+
+    private Histogram diskWaitTimeMetrics;
+    private Histogram diskServiceRateMetrics;
+    private Histogram diskUtilizationMetrics;
+    private MetricsRegistry metricsRegistry;
+    private boolean metricsInitialised;
+    private static final Logger LOG = LogManager.getLogger(RTFDisksCollector.class);
+
+    public RTFDisksCollector() {
+        super(
+                MetricsConfiguration.CONFIG_MAP.get(DisksCollector.class).samplingInterval,
+                "RTFDisksCollector",
+                StatMetrics.DISKS_COLLECTOR_EXECUTION_TIME,
+                StatExceptionCode.DISK_METRICS_COLLECTOR_ERROR);
+        this.metricsInitialised = false;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        OSMetricsGenerator generator = OSMetricsGeneratorFactory.getInstance();
+        if (generator == null) {
+            LOG.error("could not get the instance of OSMetricsGeneratorFactory class");
+            return;
+        }
+
+        metricsRegistry = OpenSearchResources.INSTANCE.getMetricsRegistry();
+        if (metricsRegistry == null) {
+            LOG.error("could not get the instance of MetricsRegistry class");
+            return;
+        }
+
+        LOG.debug("Executing collect metrics for RTFDisksCollector");
+
+        initialiseMetricsIfNeeded();
+        DiskMetricsGenerator diskMetricsGenerator = generator.getDiskMetricsGenerator();
+        diskMetricsGenerator.addSample();
+
+        recordMetrics(diskMetricsGenerator);
+    }
+
+    private void recordMetrics(DiskMetricsGenerator diskMetricsGenerator) {
```

Comment:
Please make sure the metrics and dimension names are in sync with the existing system.

Reply:
Yes. Metrics and dimension names are unchanged.